plugins/GSdx/Renderers/Common/GSTextureCache.cpp

```diff
@@ -1451,6 +1452,7 @@ GSTextureCache::Source* GSTextureCache::CreateSource(const GIFRegTEX0& TEX0, con
 		{
 			src->m_texture = m_renderer->m_dev->CreateTexture(tw, th, Get8bitFormat());
 			src->m_should_have_tex_palette = true;
+			AttachPaletteToSource(src, psm.pal, true);
```

Comment:
Not related to this line of code in particular.
Is there a possibilities that we attach the palette twice ?
We attach in Create Src and in LookUp Src.

Maybe it could be possible to avoid the compare64 check in lookup ? It isn't critical if current code is safe but potentially a bit "slower".

Reply:
it feels like the last commit, isn't it ?

Reply:
You are right, this commit alone introduces double palette attachment (as it invalidates the current fast path), but with the last commit the fast path is managed in an easier way and it always avoid double attachment in Create Src and LookUp Src.

Reply:
I think the best will be to do a single commit
* that attach palette in all cases / as soon as possible
* the bool new_source of latest commit (which I think is the fix to double attachment)

This way we have an atomic (and logical) commit.

Reply:
I agree with that, the intermediate commit breaks fast path so it's better to squash them in one.